pkg/controllers/user/monitoring/projectHandler.go

```diff
@@ -64,17 +64,23 @@ func (ph *projectHandler) doRevertSync(project *mgmtv3.Project, clusterName stri
 		return nil
 	}
 
-	// drop metrics expressions from management cluster
-	if !mgmtv3.ProjectConditionMetricExpressionDeployed.IsFalse(project) {
+	// for cluster monitoring, still need to create metrics expressions per projects into management cluster
+	_, err := mgmtv3.ProjectConditionMetricExpressionDeployed.DoUntilTrue(project, func() (runtime.Object, error) {
+		projectName := fmt.Sprintf("%s:%s", project.Spec.ClusterName, project.Name)
+
 		for _, graph := range preDefinedProjectGraph {
-			err := ph.app.cattleProjectGraphClient.DeleteNamespaced(project.Name, graph.Name, &metav1.DeleteOptions{})
-			if err != nil && !apierrors.IsNotFound(err) {
-				return errors.Wrap(err, "failed to withdraw metric expression")
+			newObj := graph.DeepCopy()
+			newObj.Namespace = project.Name
+			newObj.Spec.ProjectName = projectName
+			if _, err := ph.app.cattleProjectGraphClient.Create(newObj); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
I don't want to do this logic in `doRevertSync` function. Please create a new function for it.